### PR TITLE
Adding directions for secondary region and replication

### PR DESCRIPTION
**Pull request by bcaton85**
Adds directions for setting up a secondary environment as well as adds replication to the terraform scripts.

### CODE REVIEW

File: infrastructure/README.md [hunk 1]
```diff
@@ -18,75 +18,184 @@ Before running the infra scripts, you need to:
...
+    ```
+
+## Creating the secondary instance
+
+**NOTE** Creating the secondary instance requires completing the setup of the primary instance from the previous section first.
+
+1. The secondary instance is readonly and requires service keys to be configured through the primary instance. The service keys are used for functions related to signing and token generation. First we need to generate those keys via a script in the Quay repository.
+
+```
+$ git clone https://github.com/quay/quay  # clone the Quay repository
```
Comment (syed) [inline infrastructure/README.md:85]: you can generate this in terraform. Just curl that file in a `null` resource and add the keys to the template

Comment (bcaton85) [reply infrastructure/README.md]: Right now there's no way to output read information from the null resource. I tried outputting to a file and reading from that but the file command expects the file to exist before terraform apply is ran. There's also the external resource but that runs on the plan and before the apply and since this runs commands against the primary DB I opted to not do that.

I instead added the service key setup to a script that can be ran and then pass the kid and pem as parameters which will get templated, to make things easier. 

There might be a way to get the parameter and reference it in the template generation but Terraform doesn't do well with running scripts like this imo.

Comment (syed) [reply infrastructure/README.md]: Here's my take
```
locals {
  prefix = "testing"
}

data "http" "genkeyscript" {
  url = "https://raw.githubusercontent.com/quay/quay/master/tools/generatekeypair.py"
}

resource "null_resource" "run_python_script" {
  provisioner "local-exec" {
    command = "python -c '${data.http.genkeyscript.body}' '${local.prefix}'"
  }
}

data "local_file" "key" {
  filename = "${local.prefix}.jwk"
}

data "local_file" "cert" {
  filename = "${local.prefix}.pem"
}

data "local_file" "key_id" {
  filename = "${local.prefix}.kid"
}

output "key" {
  value = data.local_file.key.content
}

output "cert" {
  value = data.local_file.cert.content
}

output "key_id" {
  value = data.local_file.key_id.content
}

resource "null_resource" "cleanup" {
  provisioner "local-exec" {
    command = "rm -f '${local.prefix}.jwk' '${local.prefix}.pem' '${local.prefix}.kid'"
  }
  depends_on = [data.local_file.key, data.local_file.cert, data.local_file.key_id]
}
```
save this in a file `main.tf` and run `terraform apply`

Comment (bcaton85) [reply infrastructure/README.md]: Yup got it to work with the `data.local_file` resource, updated.

File: infrastructure/setup_service_keys.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
+set -euo pipefail
```
Comment (syed) [inline infrastructure/setup_service_keys.sh:1]: is this file needed?

File: infrastructure/update_service_keys.sql.template [hunk 3]
```diff
@@ -0,0 +1,14 @@
+INSERT INTO servicekey (name, service, metadata, kid, jwk, created_date, expiration_date) 
+    VALUES ('quay-readonly', 
+                'quay', 
+                '{}', 
+                'QUAY_READONLY_KID',   
+                'QUAY_READONLY_JWK',
+                '2024-01-01 00:00:00', 
+                '2025-01-01 00:00:00');
```
Comment (syed) [inline infrastructure/update_service_keys.sql.template:8]: can we make expiry far far into the future like `3024`
